src/Mod/Measure/Gui/ViewProviderMeasureBase.cpp

```diff
@@ -0,0 +1,577 @@
+{
+    std::string propName = prop.getName();
+    if (propName == "Visibility") {
+        if (!docObj.Visibility.getValue()) {
+            // show ourselves only if subject is visible
+            setVisible(false);
+        } else {
+            // here, we don't know if we should be visible or not, so we have to check the whole subject
+            setVisible(isSubjectVisible());
+        }
+    }
+}
+
+
+//NOLINTBEGIN
+PROPERTY_SOURCE(MeasureGui::ViewProviderMeasure, MeasureGui::ViewProviderMeasureBase)
+//NOLINTEND
+
+//! the general purpose view provider.  handles area, length, etc - any measure without a specialized VP
+ViewProviderMeasure::ViewProviderMeasure()
+{
+    sPixmap = "umf-measurement";
+
+
+    // setupSceneGraph for leader?
+    const size_t lineCount(3);
+
+    // indexes used to create the edges
+    // this makes a line from verts[0] to verts[1]
+    static const int32_t lines[lineCount] =
+    {
+        0,1,-1
+    };
+
+    pCoords = new SoCoordinate3();
+    pCoords->ref();
+
+    // Combine coordinates from baseTranslation and labelTranslation
+    auto engineCat = new SoConcatenate(SoMFVec3f::getClassTypeId());
+    auto origin = new SoSFVec3f();
+    origin->setValue(0,0,0);
+    engineCat->input[0]->connectFrom(origin);
+    engineCat->input[1]->connectFrom(&pLabelTranslation->translation);
+    pCoords->point.setNum(engineCat->output->getNumConnections());
+    pCoords->point.connectFrom(engineCat->output);
+
+    pLines  = new SoIndexedLineSet();
+    pLines->ref();
+    pLines->coordIndex.setNum(lineCount);
+    pLines->coordIndex.setValues(0, lineCount, lines);
+
+    auto lineSep = pLineSeparator;
+    lineSep->addChild(pCoords);
+    lineSep->addChild(pLines);
+    auto points = new SoMarkerSet();
+    points->markerIndex = Gui::Inventor::MarkerBitmaps::getMarkerIndex("CROSS",
+            Gui::ViewParams::instance()->getMarkerSize());
+    points->numPoints=1;
+    lineSep->addChild(points);
+
+    // Connect dragger local orientation to view orientation
+    auto vp = dynamic_cast<Gui::View3DInventor*>(Gui::Application::Instance->activeDocument()->getActiveView());
+    if (vp) {
+        Gui::View3DInventorViewer* viewer = vp->getViewer();
+        auto renderManager = viewer->getSoRenderManager();
+        auto cam = renderManager->getCamera();
+        pDraggerOrientation->rotation.connectFrom(&cam->orientation);
+    }
+
+
+}
+
+ViewProviderMeasure::~ViewProviderMeasure()
+{
+    pCoords->unref();
+    pLines->unref();
+}
+
+void ViewProviderMeasure::positionAnno(const Measure::MeasureBase* measureObject) {
+    (void)measureObject;
+
+    // Initialize the text position
+    Base::Vector3d textPos = getTextPosition();
+    auto srcVec = SbVec3f(textPos.x, textPos.y, textPos.z);
+
+    // Translate the position by the local dragger matrix (pDraggerOrientation) 
+    auto view = dynamic_cast<Gui::View3DInventor*>(Gui::Application::Instance->activeDocument()->getActiveView());

```

Comment:
Several issues:
* Avoid using the active document but instead use the document this view provider is part of. E.g. use ViewProviderDocumentObject::getActiveView()
* If the active view is not a 3d view then `view` is null and will segfault